### PR TITLE
Ensure DD_SERVICE is not used for external integrations

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
This PR is part 1 of 2 where we confirm the behaviour of service names: we want only non-external integrations to use `DD_SERVICE` (or `config.service`).

The second PR will add assertions that non-external integrations all use DD_SERVICE if it is specified by the user.

### CODE REVIEW

File: ddtrace/contrib/grpc/patch.py [hunk 1]
```diff
@@ -15,8 +15,6 @@
     distributed_tracing_enabled=True,
 ))
 
-# TODO[tbutt]: keeping name for client config unchanged to maintain backwards
-# compatibility but should change in future
```
Comment (brettlangdon) [inline ddtrace/contrib/grpc/patch.py]: any reason to remove this? seems unrelated to this PR

File: tests/base/__init__.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 import ddtrace
 
+from ..subprocesstest import SubprocessTestCase
 from ..utils import override_env
 from ..utils.tracer import DummyTracer
 from ..utils.span import TestSpanContainer, TestSpan, NO_CHILDREN
 
 
-class BaseTestCase(unittest.TestCase):
+class BaseTestCase(SubprocessTestCase):
```
Comment (brettlangdon) [inline tests/base/__init__.py]: should everything be a subprocess test?

File: tests/contrib/aiopg/test.py [hunk 3]
```diff
@@ -198,6 +199,26 @@ def test_patch_unpatch(self):
         assert spans, spans
         assert len(spans) == 1
 
+    @run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
```
Comment (brettlangdon) [inline tests/contrib/aiopg/test.py]: Do we need to do this as a subprocess test?

File: ddtrace/contrib/grpc/patch.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/grpc/patch.py]: I thought @majorgreys was alluding to these future `DD_SERVICE` changes.. in which case the future _is now_ :smile: If not I can put it back if we have other plans for the client service name.

File: tests/base/__init__.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply tests/base/__init__.py]: Subclassing `SubprocessTestCase` won't run tests in a subprocess unless they are marked with `@run_in_subproccess`. Else they're just plain-old unittest test cases.

File: ddtrace/contrib/grpc/patch.py (hunk 1 above)
Comment (brettlangdon) [reply ddtrace/contrib/grpc/patch.py]: oh, it should still change in the future :)

File: tests/base/__init__.py (hunk 2 above)
Comment (brettlangdon) [reply tests/base/__init__.py]: got it

File: tests/contrib/aiopg/test.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply tests/contrib/aiopg/test.py]: We don't _need_ to.. it's just a bit more elegant than using testing utils. 🤷‍♂

File: tests/base/__init__.py (hunk 2 above)
Comment (brettlangdon) [reply tests/base/__init__.py]: think it might be worth adding `run_in_subprocess` as a static method on `SubprocessTestCase`?

Thinking:

```python
class MyTestCase(BaseTestCase):
    @BaseTestCase.run_in_subprocess()
    def test_my_test(self):
        pass
```

To keep down on import fatigue.

Not actually a big deal, was just a thought

File: tests/contrib/aiopg/test.py (hunk 3 above)
Comment (brettlangdon) [reply tests/contrib/aiopg/test.py]: If we have tests that assert that `DD_SERVICE="mysvc"` -> `config.service = "mysvc", then we probably don't need to test via env variables here, we can probably just have:

```python
def test_service(self):
    with override_global_config(service="mysvc"):
        # generate spans
        spans = self.get_spans()
        assert spans[0].service != "mysvc"
```

Comment (brettlangdon) [reply tests/contrib/aiopg/test.py]: if the `with override_global_config():` is starting to be a huge pain, we can figure out a better way.

Maybe there is a way via pytest fixtures, or maybe we can have a function wrapper or something (e.g. `override_global_config` returns an instance of a class which defines both `__enter__`/`__exit__` as well as `__call__`, if `__call__` is used then it is treated as a decorator which uses itself as a context manager to call the wrapped function.

File: tests/base/__init__.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply tests/base/__init__.py]: oh, good idea! I was thinking about that with all the additional imports.

File: tests/contrib/cassandra/test.py [hunk 4]
```diff
@@ -459,3 +460,36 @@ def test_backwards_compat_get_traced_cassandra():
     cluster = get_traced_cassandra()
     session = cluster(port=CASSANDRA_CONFIG['port']).connect()
     session.execute('drop table if exists test.person')
+
+
+class TestCassandraConfig(SubprocessTestCase):
```
Comment (brettlangdon) [inline tests/contrib/cassandra/test.py]: we should do `BaseTestCase` here?

File: tests/contrib/aiopg/test.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply tests/contrib/aiopg/test.py]: I guess I just have an internal bias against `override_global_config`, not because of the ergonomics, but because I don't really trust it to perfectly emulate what happens in reality. It's a test helper that is written to perform the same side-effects as the expected real-world usage.

It's maybe a bit irrational of me..  but I like the idea of not having to depend on yet another piece of code that we wrote in order to test our functionality 😛.

**Comment by Kyle-Verhoog** [comment]
@brettlangdon I'm going to merge this just so we can progress with the other DD_SERVICE PR(s).

**Comment by brettlangdon** [comment]
Ok, we can talk about testing as a separate thing. Shouldn't block this.
